server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/state/TransportEnsureClusterStateVersionAppliedActionIT.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.action.admin.cluster.state;
+
+import org.elasticsearch.ElasticsearchTimeoutException;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateUpdateTask;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.IndexVersion;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.util.Locale;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicLong;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0, numClientNodes = 0)
+public class TransportEnsureClusterStateVersionAppliedActionIT extends ESIntegTestCase {
+
+    public void testVersionThatIsAlreadyApplied() {
+        var masterNode = internalCluster().startMasterOnlyNode();
+        var node1 = internalCluster().startNode();
+        ensureStableCluster(2);
+
+        String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        createIndex(indexName);
```

Comment:
It's pretty, but maybe worth a comment? I think what you're doing is creating an index because that causes a cluster state update, and then using green status as proof that both nodes have applied the update, on the assumption that the index is created with 1 shard and 1 replica?

I'd probably also use the explicit `createIndex(indexName, 1, 1)` if that's the case.